### ISSUE TITLE
Fallback

### ISSUE THREAD

**Issue by corporateanon** [kingjan1999/platform-folders#2]
What happens when it fails to get a folder? Does it throw an error or return null or return some fallback path?

**Comment by kingjan1999** [comment]
Basically we use the [sago007/PlatformFolders](https://github.com/sago007/PlatformFolders) error routines here, especially with regard to possible fallbacks.

In some cases, however, I messed around when creating this library, which could theoretically lead to exceptions that occur being ignored or even causing a crash.

I openly admit that this behavior is certainly not reasonable and I will improve it in the next release.

**Comment by corporateanon** [comment]
Thanks for your quick response!

I am developing a cross-platform application with a download feature, and i
want to provide a fallback option for platforms not supported by this
library. Though i haven't had a chance yet to reproduce any error, I will
try/catch just in case.

Did you test it in headless Linux distributions?

пн, 16 апр. 2018, 20:52 Jan <notifications@github.com>:

**Comment by kingjan1999** [comment]
I've just published v0.2.1 on npm, which now includes a sensible error handling. If something strange happens, it'll now always throw a js-error.

It should work on headless linux dists ([as shown here](https://travis-ci.org/kingjan1999/platform-folders/builds/367336474)). While developing, I noticed that the Mac builds are failing, I guess this is an upstream bug with [sago007/PlatformFolders](https://github.com/sago007/PlatformFolders).
